pkg/ingester/ingester.go

```diff
@@ -954,6 +963,10 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReques
 	// retain anything from `req` past the exit from this function.
 	defer cleanUp()
 
+	if !i.cfg.EnablePushAPI {
```

Comment:
We need to allow `PushWithCleanup()` because that's what we call to ingest data from Kafka (see `NewPartitionReaderForPusher()` called from `ingester.New()`).

Why not existing test on ingest storage failed (even if we have very few, at least one should have failed). Reason is that we're not disabling the Push API in ingest storage tests. Please set `EnablePushAPI` to `false` in `createTestIngesterWithIngestStorage()`. You should get at least `TestIngester_QueryStream_IngestStorageReadConsistency` failing.

So how to solve it? We could be tempted to remove this check and keep `StartPushRequest()`, but `StartPushRequest()` is called if `LimitInflightRequestsUsingGrpcMethodLimiter` is disabled (and BTW we need to keep `LimitInflightRequestsUsingGrpcMethodLimiter` disabled with ingest storage because there's no ingestion via gRPC).

I think to solve it easily (hopefully) we have two move the check out of `StartPushRequest()` and `PushWithCleanup()` into ` Push()`, move the current `Push()` content into `push()` and then call `push()` when we ingest data via Kafka (or something similar, you got the idea).

Reply:
I decided to try re-working this differently: instead of disabling the functionality from the inside of `Ingester`, I do that in `pkg/mimir` (see the changes below), closer to where we register an implementation of the `api.Ingester` (the actual gRPC handlers) in the gRPC server.

I feel this will be less prone to human errors. What do you think? 

Note I haven't updated the tests, yet; just wanted to get feedback, first

cc @pstibrany

Reply:
Wrapping ingester to disable its `Push` gRPC method sounds good to me. This way the test changes that Marco asked for in his comment will not be easy to do, but I don't think that's a blocker.

Let's please move that `apiIngester` type into `pkg/ingester` (I'd also recommend finding a better name for it, eg. `ingesterWithDisabledPush`?), and minimize unrelated changes in this PR.